README.md

```diff
@@ -162,3 +162,216 @@ ARGS:
+sub_dir/more_dir/not_file
+sub_dir/more_dir/test_file_six
+sub_dir/test_file_five
+sub_dir/test_file_three
+sub_dir/test_four
+test_file_one
+test_file_two
+test_one
+this_is_a_test
+```
+
+`fd` is magic, it will look for a `.gitignore` file and treat the rules inside it as rules in the search pattern. So if we have a `.gitignore` file like: 
+
+```
+*.sh
+```
+`fd` will then never look for any files that end in `.sh`. We can tell `fd` to ignore `.gitignore` files with `-I` (or `--ignore`) we can temporarliy stop that from happening: 
+
+`fd -I me`
+
+```
+sub_dir/more_dir/even_further_down/not_me.sh
+```
+
+Searching for a file extension is easy too, don't forget your regex rules (`$` is the end of a line): 
+
+`fd -I '\.sh$'`
+
+```
+sub_dir/more_dir/even_further_down/not_me.sh
+```
+
+What if we wanted to run some complicated bash follow on to the files? `xargs` can help us with that: 
+
+`fd -0 'test' | xargs -0 -I {} cp {} {}.new`
+
+In the example there's a couple things to make note:
+  - First we are telling `fd` we want a null character to seperate the files `-0`, this is important when passing to `xargs`
+  - second we are piping the output to `xargs` and telling this program to expect input null terminated with `-0` (the same syntax that `fd` was built with).
+  - Then for fun are are using `-I` to replace a string `{}` and lauching `cp` to copy the file `{}` to a file ending in `{}.new` 
+
+That gives us: 
+
+```
+.
+├── .gitignore
+├── not_file
+├── sub_dir
+│   ├── .here_be_tests
+│   ├── more_dir
+│   │   ├── .not_here
+│   │   ├── even_further_down
+│   │   │   ├── not_me.sh
+│   │   │   ├── test_seven
+│   │   │   ├── test_seven.new
+│   │   │   ├── testing_eight
+│   │   │   └── testing_eight.new
+│   │   ├── not_file -> /Users/fd_user/fd_examples/not_file
+│   │   ├── test_file_six
+│   │   └── test_file_six.new
+│   ├── test_file_five
+│   ├── test_file_five.new
+│   ├── test_file_four
+│   ├── test_file_four.new
+│   ├── test_file_three
+│   └── test_file_three.new
+├── test_file_one
+├── test_file_one.new
+├── test_file_two
+├── test_file_two.new
+├── test_one
+├── test_one.new
+├── this_is_a_test
+└── this_is_a_test.new
+```
+
+`fd` can also show us the absolute path vs. the full path: 
+
+`fd -a 'new'`
+
+```
+/Users/fd_user/fd_examples/sub_dir/more_dir/even_further_down/test_seven.new
+/Users/fd_user/fd_examples/sub_dir/more_dir/even_further_down/testing_eight.new
+/Users/fd_user/fd_examples/sub_dir/more_dir/test_file_six.new
+/Users/fd_user/fd_examples/sub_dir/test_file_five.new
+/Users/fd_user/fd_examples/sub_dir/test_file_four.new
+/Users/fd_user/fd_examples/sub_dir/test_file_three.new
+/Users/fd_user/fd_examples/test_file_one.new
+/Users/fd_user/fd_examples/test_file_two.new
+/Users/fd_user/fd_examples/test_one.new
+/Users/fd_user/fd_examples/this_is_a_test.new
+```
+
+Be careful when using the full path flag (`--full-path`), as this searches the **path** of the file as well so regex like `^` won't work as intended: 
```

Comment:
I would rather like to advertise `--full-path` as a feature, not as a hazard :smile:. For example:
```
> fd -p 'http.*log' /var/log
/var/log/httpd/access_log
/var/log/httpd/error_log
```